src/Seabug/CallContract.purs

```diff
@@ -73,26 +74,27 @@ import Serialization.Hash
   )
 import Types.BigNum as BigNum
 import Types.Natural as Nat
-import Wallet (Wallet, mkNamiWalletAff, mkGeroWalletAff)
+import Types.RawBytes (rawBytesToHex)
 
-callConnectWallet
-  :: forall (r :: Row Type)
-   . String
-  -> Effect (Promise Wallet)
-callConnectWallet walletOption = case walletOption of
-  "Nami" -> Promise.fromAff mkNamiWalletAff
-  "Gero" -> Promise.fromAff mkGeroWalletAff
-  _ -> throwError <<< error $ "Unsupported wallet: " <> walletOption
+defaultWalletSpec :: Maybe WalletSpec
+defaultWalletSpec = Just ConnectToNami
+
+callGetWalletPkh :: ContractConfiguration -> Effect (Promise (Nullable String))
+callGetWalletPkh cfg = Promise.fromAff do
+  contractConfig <- liftEither $ buildContractConfig cfg defaultWalletSpec
+  (toNullable <<< map (rawBytesToHex <<< ed25519KeyHashToBytes <<< unwrap)) <$>
+    runContract contractConfig ownPubKeyHash
 
 callGetWalletBalance
-  :: ContractConfiguration -> Effect (Promise (Nullable Value))
+  :: ContractConfiguration -> Effect (Promise (Nullable BigInt))
 callGetWalletBalance cfg = Promise.fromAff do
```

Comment:
This might be better as getWalletAda or getWalletLovelace
getBalance in other cardano contexts often means the full Value